### PR TITLE
[ML][Data frame] make sure that fields exist when creating progress

### PR DESCRIPTION
**Pull request by benwtrent**
Since we do not yet support `missing_bucket` in our pivot transform, the default behavior in Composite aggregations is to skip that bucket. This causes our total doc count to be inaccurate when there are missing/null fields. To address this, an `exists` query for each of the pivoted fields needs to be done when `missing_bucket: false` (currently, always the case).

### CODE REVIEW

File: x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/transforms/TransformProgressGatherer.java [hunk 1]
```diff
@@ -49,10 +51,20 @@ public static void getInitialProgress(Client client,
     public static SearchRequest getSearchRequest(DataFrameTransformConfig config) {
         SearchRequest request = new SearchRequest(config.getSource().getIndex());
         request.allowPartialSearchResults(false);
+        BoolQueryBuilder existsClauses = QueryBuilders.boolQuery();
+        config.getPivotConfig()
```
Comment (przemekwitek) [inline x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/transforms/TransformProgressGatherer.java]: [optional]
IMO in this case the old good foreach syntax is both shorter and more readable:
for (SingleGroupSource src : config.getPivotConfig().getGroupConfig().getGroups().values()) {
  existsClauses.must(QueryBuilders.existsQuery(src.getField())));
}

But leaving the choice up to you.

File: x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/transforms/DataFrameTransformPersistentTasksExecutor.java [hunk 2]
```diff
@@ -131,7 +131,7 @@ protected void nodeOperation(AllocatedPersistentTask task, @Nullable DataFrameTr
                     .setInitialPosition(stateAndStats.getTransformState().getPosition())
                     .setProgress(stateAndStats.getTransformState().getProgress())
                     .setIndexerState(currentIndexerState(stateAndStats.getTransformState()));
-                logger.info("[{}] Loading existing state: [{}], position [{}]",
+                logger.debug("[{}] Loading existing state: [{}], position [{}]",
```
Comment (przemekwitek) [inline x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/transforms/DataFrameTransformPersistentTasksExecutor.java]: Double-checking: is this change intended in this PR?

File: x-pack/plugin/data-frame/qa/single-node-tests/src/test/java/org/elasticsearch/xpack/dataframe/integration/DataFrameTransformProgressIT.java [hunk 3]
```diff
@@ -163,6 +161,23 @@ public void testGetProgress() throws Exception {
         assertThat(progress.getRemainingDocs(), equalTo(35L));
         assertThat(progress.getPercentComplete(), equalTo(0.0));
 
+        histgramGroupConfig = new GroupConfig(Collections.emptyMap(),
```
Comment (przemekwitek) [inline x-pack/plugin/data-frame/qa/single-node-tests/src/test/java/org/elasticsearch/xpack/dataframe/integration/DataFrameTransformProgressIT.java]: Does this method cover multiple independent test case scenarios?
How hard would it be to split it into a few independent test cases?

File: x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/transforms/DataFrameTransformPersistentTasksExecutor.java (hunk 2 above)
Comment (benwtrent) [reply x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/transforms/DataFrameTransformPersistentTasksExecutor.java]: @przemekwitek yeah, this change was skipped in a previous PR and I just put it in this one. Low impact and did not want to have to kick of an entire CI run on my previous PR because of a logging level change.

File: x-pack/plugin/data-frame/qa/single-node-tests/src/test/java/org/elasticsearch/xpack/dataframe/integration/DataFrameTransformProgressIT.java (hunk 3 above)
Comment (benwtrent) [reply x-pack/plugin/data-frame/qa/single-node-tests/src/test/java/org/elasticsearch/xpack/dataframe/integration/DataFrameTransformProgressIT.java]: @przemekwitek not too difficult, but the concern is creating and destroying indices, and speed.

**Review by hendrikmuhs — APPROVED** [review]
LGTM

**Comment by hendrikmuhs** [comment]
For the record: adding `missing_bucket` is tracked in #42941

File: x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/transforms/DataFrameTransformPersistentTasksExecutor.java (hunk 2 above)
Comment (przemekwitek) [reply x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/transforms/DataFrameTransformPersistentTasksExecutor.java]: No problem, just wanted to make sure that it is not a leftover from your local debugging.
